guides/development-workflow.md

```diff
@@ -0,0 +1,284 @@
+  3. merge the pull request
+  4. manually merge the pull request back into develop
+  ```
+  git checkout develop
+  git pull
+  git merge origin/master
+  git push
+  ```
+  5. move the card into 'Deployed to production'
+
+---
+
+## Branching
+
+#### How to make a branch
+
+Create the branch off of develop:
+```
+git fetch
+git checkout develop
+git branch <branch-name>
+git checkout <branch-name>
+```
+
+**Naming the branch**
+
+Having a convention allows for:
+
+* all branches connected to a story or ticket are easily connected now and in the future
+* others can easily identify the intent of the contents
+
+**Convention:**
+
+* Feature - A branch that adds a new feature, as defined by the specified story
+* Fix - A branch that corrects a problem in a feature already merged into develop
+* Chore - Chore branches are used for routine tasks or tickets which are not emergencies
+* Hotfix - A branch that adds an urgent fix to a problem that affects production. These branches are based on master and do not go via develop, so must be also be merged into develop when deployed
+
+```
+feature/{STORY ID}-{CONCISE TITLE}
+fix/{STORY/TICKET ID}-{CONCISE TITLE}
+chore/{STORY ID}-{CONCISE TITLE}
+hotfix/{TICKET ID}-{CONCISE TITLE}
+```
+
+**Examples:**
+```
+feature/523797477-add-logging-to-registration
+fix/423797477-logging-happens-in-both-environments
+chore/1934729239-reduce-caching-for-contact-details
+hotfix/253947623-remove-breaking-change-to-repair-creation
+```
+
+#### Update your branch with the latest changes on develop (rebasing)
+
+When working in a team other pull requests will often get merged in before yours is finished  It's important to make sure you're developing against the latest version of the code base as much as possible.
+
+You can do this by rebasing your work on top of the develop branch.
+
+```
+git checkout <branch-name>
+git rebase origin/develop
+git push --force-with-lease
+```
+
+During the rebase there may be conflicts which should be resolved at this point. If in doubt talk to the other developer who you're conflicting with and reason out what the correct state should be.
+
+Forcing pushing in this situation is the only time it should be necessary, adding `--force-with-lease` ensures you don't accidentally overwrite someone else's additions.
+
+## Committing
+
+Writing a good commit message serves three important purposes.
+
+* to speed up the review process
+* to encourage individual commits to be well small, bounded and concise
+* to help future maintainers (it could be you) in say 3 years time, to find out why a particular change was made and what context was relevant at the time
+
+**Convention:**
+```
+(feature): when we are adding something new
+(fix): when we are fixing something existing
+(refactor): when we are changing the implementation of something existing without changing its behaviour
+(chore): catch all for when none of the above apply and there is no immediate user need eg. gem/package updates
```

Comment:
I'm still not convinced of the utility of these little markers at the start of a commit message.

What does it matter, 2 years into the future, whether a commit you're looking at is a feature or a fix? What matters is that the commit message accurately and clearly describes what the commit does, and why it was done. And if you're adding `(feature): ` to the start of your commit message then you're taking up 11 characters that could otherwise be spent on a succinct explanation.